types/context.go

```diff
@@ -175,6 +173,7 @@ func (c Context) WithHeaderHash(hash []byte) Context {
 
 // WithBlockTime returns a Context with an updated CometBFT block header time in UTC with no monotonic component.
 // Stripping the monotonic component is for time equality.
+// Deprecated: use WithHeaderInfo instead
 func (c Context) WithBlockTime(newTime time.Time) Context {
```

Comment:
I think we should kill the setter but keep the getters.
Otherwise now, WithBlockTime/BlockTime with behave very weirdly as BlockTime gets is from the header info.